doc/examples/big_data_ingestion.py

```diff
@@ -0,0 +1,228 @@
+                pass
+
+
+###########################################################################
+# Let's run it
+# -----------------------------
+#
+# Now let’s run the data pipeline end-to-end:
+#
+# First, let's parse some arguments.
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--large-scale-test",
+    action="store_true",
+    help="Read a smaller dataset for quick testing purposes.")
+
+args, _ = parser.parse_known_args()
+
+###############################################################################
+#
+# After that, let's generate 100MB of Parquet files,
+# create the shuffle pipeline by reading those generated Parquet files,
+# and use training workers to consume the pipeline.
+
+if not args.large_scale_test:
+    NUM_TRAINING_WORKERS = 4
+    NUM_EPOCHS = 5
+    SIZE_100MB = 100 * 1024 * 1024
+
+    # create a local ray cluster.
+    ray.init()
+
+    def generate_example_files(size_bytes: int) -> str:
+        tmpdir = tempfile.mkdtemp()
+        ray.data.read_datasource(
+            RangeDatasource(), block_format='arrow',
+            n=size_bytes // 8).write_parquet(tmpdir)
+        return tmpdir
+
+    example_files_dir = generate_example_files(SIZE_100MB)
+
+    splits = create_shuffle_pipeline(example_files_dir, NUM_EPOCHS,
+                                     NUM_TRAINING_WORKERS)
+
+    training_workers = [
+        TrainingWorker.remote(rank, shard) for rank, shard in enumerate(splits)
+    ]
+
+    start = time.time()
+    # Let's run the e2e pipeline
+    ray.get([worker.train.remote() for worker in training_workers])
+    print(f"total ingestion time: {time.time() - start}")
+
+#################################################################################
+# Scale the shuffle ingestion pipeline
+# --------------------------------------------------------
+#
+# Scaling the shuffle ingestion pipeline is simple. With Ray, we can linearly
+# scale the pipeline from ingesting 100MB of data to 500GB of data by adding
+# more machines.
+#
+# To ingest 500GB of data, we'll set up a Ray Cluster.
+# The provided ``big_data_ingestion.yaml``
+# cluster config can be used to set up an AWS cluster with 70 CPU nodes and
+# 4 GPU nodes. Using following command to bring up the Ray cluster.
+#
+# .. code-block:: bash
+#
+#     $ pip install ray boto3
+#     $ ray up big_data_ingestion.yaml
+#
+# After the cluster is started, let's implement our large scale ingestion test:
+#
+# First, since we are runing on a cluster, let's create the pipeline from
+# RangeDatasource directly. In this way we don't need to set up S3 for storing
+# generated data.
+
+
+def create_large_shuffle_pipeline(data_size_bytes: int, num_epochs: int,
+                                  num_shards: int) -> List[DatasetPipeline]:
+    # TODO: reading data from distributed filesystem (such as S3 or HDFS).
+    return ray.data.read_datasource(
+            RangeDatasource(), block_format='arrow',
+            n=data_size_bytes // 8) \
```

Comment:
I think you can just `ray.data.range_arrow(n)` instead. But maybe it's better to return fake data of the same schema as the S3 data? Otherwise, it would be unrealistic / can't send to horovod.